src/Database/Redis/PubSub.hs

```diff
@@ -212,6 +238,275 @@ pubSub initial callback
                 PubSubState{..} <- get
                 unless (subCnt == 0 && pending == 0) recv
 
+-- | A Redis channel name
+type RedisChannel = ByteString
```

Comment:
I don't have a big feeling one way or the other, but for the most part I expect the channels to be raw strings with OverloadedStrings enabled so I don't know how much type saftey you would get.  I made them primarily so the haddock documentation looks nice.  But as I said, I don't really lean one way or the other.